### PR TITLE
Upgrade setuptools dependency

### PR DESCRIPTION
**Pull request by Rebits**
|Related issue|
|-------------|
|      #3760       |

## Description

Upgrade setuptools to `>65.5.1.`

### Updated

- Setuptools version

---

## Testing performed

| Tester             | Test path | Jenkins | Local  | OS | Commit | Notes                |
|--------------------|-----------|---------|--------|-----|--------|----------------------|
| @user (Developer)  |           | [:yellow_circle: ](https://ci.wazuh.info/job/Test_integration/36691/console) | ⚫⚫⚫ |         |         | Nothing to highlight |
| @user (Reviewer)   |           | ⚫⚫⚫ | :no_entry_sign: :no_entry_sign: :no_entry_sign:  |        |         | Nothing to highlight |

### CODE REVIEW

**Comment by Rebits** [comment]
### 22/05/2023

**Buid**: https://ci.wazuh.info/job/Test_integration/39035/

**Comment by Rebits** [comment]
### 28/06/2023
- https://ci.wazuh.info/job/Test_integration/40628/

File: .github/workflows/dependencies_validator.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      - deps/wazuh_testing/setup.py
+jobs:
+  validate-framework:
+    runs-on: ubuntu-20.04
+    continue-on-error: true
+    strategy:
+      matrix:
+        python-version:
+          - 3.10.9
+          - 3.7.1
```
Comment (damarisg) [inline .github/workflows/dependencies_validator.yaml:15]: The [PR](https://github.com/wazuh/wazuh-jenkins/pull/5342/files) shows that you are using Python 3.11. However, for this PR, you are using Python 3.7 and Python 3.10. Could you clarify the cause, please?

**Review by damarisg — CHANGES_REQUESTED** [review]

**Review by verdx — APPROVED** [review]
Merge blocked until https://github.com/wazuh/wazuh-jenkins/issues/5407 is closed

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,6 +8,7 @@ Wazuh commit: TBD \
 Release report: TBD
 
 ### Changed
+- Update setuptools dependency ([#3788](https://github.com/wazuh/wazuh-qa/pull/3788))
```
Comment (juliamagan) [inline CHANGELOG.md:11]: This entry is duplicated (line 11 and line 77)

File: requirements.txt [hunk 3]
```diff
@@ -17,18 +17,18 @@ numpy>=1.18.1
 pandas>=1.1.5
 pillow>=6.2.0; platform_system == "Linux" or platform_system == "Darwin" or platform_system=='Windows'
 psutil>=5.6.6
-py~=1.10.0; platform_system != "Windows" or python_version <= "3.10"
-py==1.11.0; platform_system == "Windows" and python_version >= "3.11"
+py~=1.10.0; python_version <= "3.9"
+py==1.11.0; python_version >= "3.10"
```
Comment (juliamagan) [inline requirements.txt:21]: Isn't this removing the newly added change at https://github.com/wazuh/wazuh-jenkins/issues/5407?

**Review by juliamagan — CHANGES_REQUESTED** [review]

File: requirements.txt (hunk 3 above)
Comment (verdx) [reply requirements.txt:21]: Not exactly, it is expanding it outside Windows. The other PR was only for Windows but this one has a broader scope and has been investigated as so.

**Pull request merged**